runtime/memprof.c

```diff
@@ -518,10 +665,85 @@ void caml_memprof_track_interned(header_t* block, header_t* blockend) {
       p = next_p;
     }
 
-    register_postponed_callback(
-      Val_hp(p), mt_generate_binom(next_p - next_sample_p) + 1,
-      Unmarshalled, &callstack);
-
+    if(callstack == 0) callstack = capture_callstack_postponed();
+    if(callstack == 0) break;  /* OOM */
+    new_tracked(mt_generate_binom(next_p - next_sample_p) + 1,
+                Hd_hp(p), 1, young, Val_hp(p), callstack);
```

Comment:
Note: the comment above this look is now slightly overkill, as there is no specific effort required to postpone, this is now the least-effort path with the new API.